File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/BatchSettings.java
package ru.bozaro.gitlfs.client;

import org.jetbrains.annotations.NotNull;

import static ru.bozaro.gitlfs.common.Constants.BATCH_SIZE;

/**
 * Batch settings.
 *
 * @author Artem V. Navrotskiy <bozaro@users.noreply.github.com>
 */
public class BatchSettings {
  /**
   * Maximum objects in batch request.
   */
  private int limit = BATCH_SIZE;
  /**
   * Minimum download/upload requests in queue for next batch request.
   */
  private int threshold = 10;
  /**
   * Retry on failure count.
   */
  private int retryCount = 3;

  public BatchSettings() {
  }

  public BatchSettings(int limit, int threshold, int retryCount) {
    this.limit = limit;
    this.threshold = threshold;
    this.retryCount = retryCount;
  }

  public int getLimit() {
    return limit;
  }

  @NotNull
  public BatchSettings setLimit(int limit) {
    this.limit = Math.min(limit, 1);
    return this;
  }

  public int getThreshold() {
    return threshold;
  }

  @NotNull
  public BatchSettings setThreshold(int threshold) {
    this.threshold = Math.max(threshold, 0);
    return this;
  }

  public int getRetryCount() {
    return retryCount;
  }

  @NotNull
  public BatchSettings setRetryCount(int retryCount) {
    this.retryCount = Math.max(retryCount, 1);
    return this;
  }

  @Override
  public String toString() {
    return "BatchSettings{" +
        "limit=" + limit +
        ", threshold=" + threshold +
        ", retryCount=" + retryCount +
        '}';
  }
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/BatchSettings.kt
package ru.bozaro.gitlfs.client

import ru.bozaro.gitlfs.common.Constants.BATCH_SIZE
import kotlin.math.max
import kotlin.math.min

/**
 * Batch settings.
 *
 * @author Artem V. Navrotskiy <bozaro></bozaro>@users.noreply.github.com>
 */
class BatchSettings {
    /**
     * Maximum objects in batch request.
     */
    var limit: Int = BATCH_SIZE
        private set

    /**
     * Minimum download/upload requests in queue for next batch request.
     */
    var threshold = 10
        private set

    /**
     * Retry on failure count.
     */
    var retryCount = 3
        private set

    constructor()
    constructor(limit: Int, threshold: Int, retryCount: Int) {
        this.limit = limit
        this.threshold = threshold
        this.retryCount = retryCount
    }

    fun setLimit(limit: Int): BatchSettings {
        this.limit = min(limit, 1)
        return this
    }

    fun setThreshold(threshold: Int): BatchSettings {
        this.threshold = max(threshold, 0)
        return this
    }

    fun setRetryCount(retryCount: Int): BatchSettings {
        this.retryCount = max(retryCount, 1)
        return this
    }

    override fun toString(): String {
        return "BatchSettings{" +
                "limit=" + limit +
                ", threshold=" + threshold +
                ", retryCount=" + retryCount +
                '}'
    }
}
